domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/service/ReviewScheduleService.kt

```diff
@@ -0,0 +1,135 @@
+package uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.service
+
+import mu.KotlinLogging
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewSchedule
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewScheduleNotFoundException
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.ReviewScheduleStatus
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.review.dto.UpdateReviewScheduleStatusDto
+import java.time.LocalDate
+
+private const val EXEMPTION_ADDITIONAL_DAYS = 5L
+private const val EXCLUSION_ADDITIONAL_DAYS = 10L
+private const val SYSTEM_OUTAGE_ADDITIONAL_DAYS = 5L
+
+private val log = KotlinLogging.logger {}
+class ReviewScheduleService(private val reviewSchedulePersistenceAdapter: ReviewSchedulePersistenceAdapter) {
+
+  private val statusTransitionValidator = StatusTransitionValidator()
+
+  fun updateLatestReviewScheduleStatus(
+    prisonNumber: String,
+    prisonId: String,
+    newStatusString: String,
+  ) {
+    val newStatus = ReviewScheduleStatus.valueOf(newStatusString)
+    val reviewSchedule = reviewSchedulePersistenceAdapter.getLatestReviewSchedule(prisonNumber)
+      ?: throw ReviewScheduleNotFoundException(prisonNumber)
+
+    // Validate the status transition
+    statusTransitionValidator.validate(prisonNumber, reviewSchedule.scheduleStatus, newStatus)
+
+    when {
+      newStatus == ReviewScheduleStatus.EXEMPT_SYSTEM_TECHNICAL_ISSUE -> {
+        handleSystemException(reviewSchedule, newStatus, prisonId, prisonNumber)
+      }
+      newStatus.isExemptionOrExclusion() -> {
+        updateExemptStatus(reviewSchedule, newStatus, prisonId, prisonNumber)
+      }
+      else -> {
+        updateScheduledStatus(reviewSchedule, newStatus, prisonId, prisonNumber)
+      }
+    }
+  }
+
+  private fun updateExemptStatus(
+    reviewSchedule: ReviewSchedule,
+    newStatus: ReviewScheduleStatus,
+    prisonId: String,
+    prisonNumber: String,
+  ) {
+    reviewSchedulePersistenceAdapter.updateReviewScheduleStatus(
+      UpdateReviewScheduleStatusDto(
+        reviewSchedule.reference,
+        newStatus,
+        prisonId,
+        prisonNumber = prisonNumber,
+      ),
+    )
+    performFollowOnEvents(prisonNumber, newStatus, prisonId)
+  }
+
+  private fun updateScheduledStatus(
+    reviewSchedule: ReviewSchedule,
+    newStatus: ReviewScheduleStatus,
+    prisonId: String,
+    prisonNumber: String,
+  ) {
+    val newReviewDate = calculateNewReviewDate(reviewSchedule)
+    reviewSchedulePersistenceAdapter.updateReviewScheduleStatus(
+      UpdateReviewScheduleStatusDto(
+        reviewSchedule.reference,
+        ReviewScheduleStatus.SCHEDULED,
+        prisonId,
+        latestReviewDate = newReviewDate,
+        prisonNumber = prisonNumber,
+      ),
+    )
+    performFollowOnEvents(prisonNumber, newStatus, prisonId, newReviewDate)
+  }
+
+  private fun handleSystemException(
```

Comment:
Hmmm, I wonder if we can come up with a better name for this .... it took me quite a while to work out that it was not a (JVM) exception handler! 🤣 

How about something like `updateReviewScheduleFollowingSystemTechnicalIssue` or similar?

Also, I wonder it it would be easier to grok if the method sig did not include `newStatus: ReviewScheduleStatus`, and that `ReviewScheduleStatus.EXEMPT_SYSTEM_TECHNICAL_ISSUE` was used within the method instead?